pkg/controllers/provisioning/provisioner.go

```diff
@@ -212,7 +212,11 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNod
 		return nil, err
 	}
 	nodePoolList.Items = lo.Filter(nodePoolList.Items, func(n v1beta1.NodePool, _ int) bool {
-		return n.DeletionTimestamp.IsZero()
+		vaildNodePool := n.Spec.Template.Spec.Validate()
+		if vaildNodePool != nil {
+			logging.FromContext(ctx).Errorf("Nodepool %s does not contain valid Taint or StartUpTaint , %s", n.Name, err)
```

Comment:
```suggestion
			logging.FromContext(ctx).With("nodepool", n.Name).Errorf("nodepool failed validation, %s", err)
```

I'm assuming that the internals of the validation function should give the specifics of how it failed. The caller shouldn't know the internal details of how the validation on the NodePool failed

Also, logging should be structured. If you find yourself adding structured parameters into the message, then we should probably be using a `With(...)` statement